pkg/util/version_history.go

```diff
@@ -54,6 +68,16 @@ func logVersionHistoryToFile(versionHistoryFilePath, agentVersion string, timest
 		Timestamp: timestamp,
 	}
 
+	installInfo, err := os.ReadFile(installInfoFilePath)
+	if err != nil {
+		log.Infof("Cannot read %s: %s", installInfoFilePath, err)
+	}
+	if !errors.Is(err, os.ErrNotExist) {
+		if err := yaml.Unmarshal(installInfo, &newEntry); err != nil {
+			log.Infof("Cannot deserialize yaml file: %s: %s", installInfoFilePath, err)
+		}
+	}
```

Comment:
```suggestion
	installInfo, err := os.ReadFile(installInfoFilePath)
	if err != nil {
		log.Infof("Cannot read %s: %s", installInfoFilePath, err)
	} else {
		if err := yaml.Unmarshal(installInfo, &newEntry); err != nil {
			log.Infof("Cannot deserialize yaml file: %s: %s", installInfoFilePath, err)
		}
	}
```
Regardless of the specific error, we won't be able to unmarshal anything if we couldn't read it.

Reply:
https://github.com/DataDog/datadog-agent/pull/11595/commits/fae21f75f4332a15c589714b78337d12a58dbee7